### PR TITLE
Add toResolve/toReject custom matchers

### PR DESCRIPTION
**Pull request by benjaminkay93**
### What
Added custom matcher's for determining if a promise has resolved or rejected

### Why
As pair https://github.com/jest-community/jest-extended/issues/130

### Housekeeping

- [x] Unit tests
- [x] Documentation is up to date
- [x] No additional lint warnings
- [x] Add yourself to contributors list (`yarn contributor`)
- [x] [Typescript definitions](https://github.com/jest-community/jest-extended/blob/master/types/index.d.ts) are added/updated where relevant

### CODE REVIEW

File: src/matchers/toReject/predicate.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export default async promise => {
```
Comment (mattphillips) [inline src/matchers/toReject/predicate.js]: I don't think the `async` is needed here as we aren't awaiting anything. Also the `resolved`/`rejected` values aren't being used, how about something like:

```js
export default promise => promise.then(() => false, () => true);
```

File: src/matchers/toReject/predicate.test.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import predicate from './predicate';
+
+describe('toResolve predicate', () => {
```
Comment (mattphillips) [inline src/matchers/toReject/predicate.test.js]: Should be `toReject predicate` 😄

File: src/matchers/toReject/index.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { matcherHint } from 'jest-matcher-utils';
+
+import predicate from './predicate';
+
+const passMessage = () => () =>
```
Comment (mattphillips) [inline src/matchers/toReject/index.js]: No need for the extra function here or below for the `failMessage`

File: src/matchers/toResolve/index.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { matcherHint } from 'jest-matcher-utils';
+
+import predicate from './predicate';
+
+const passMessage = () => () =>
```
Comment (mattphillips) [inline src/matchers/toResolve/index.js]: Same here

File: src/matchers/toResolve/predicate.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+export default async promise => {
```
Comment (mattphillips) [inline src/matchers/toResolve/predicate.js]: Same as above comment :)

```js
export default promise => promise.then(() => true, () => false);
```

**Comment by mattphillips** [comment]
@benjaminkay93 overall this looks awesome dude! Just a few minor comments above 😃

File: types/index.d.ts [hunk 6]
```diff
@@ -283,6 +283,16 @@ declare namespace jest {
      */
     toBeSealed(): R;
 
+    /**
+     * Use `.toResolve` when checking if a promise resolves.
+     */
+    toResolve(): R;
```
Comment (mattphillips) [inline types/index.d.ts]: Would you be able to add these types to the `expect` interface too (asymmetric matchers)

**Comment by mattphillips** [comment]
Thanks @benjaminkay93 :)

File: types/index.d.ts (hunk 6 above)
Comment (benjaminkay93) [reply types/index.d.ts]: aha I will have a nosey at the expect now, thanks matt 🎉

**Pull request merged**